CMakeLists.txt

```diff
@@ -57,49 +60,193 @@ message(STATUS EXTRA_LIBRARIES: ${EXTRA_LIBRARIES})
 set(Boost_NO_BOOST_CMAKE 1)
 set(Boost_USE_MULTITHREADED ON)
 find_package(Boost 1.58 QUIET REQUIRED COMPONENTS chrono date_time filesystem program_options regex serialization wserialization system thread)
-message(STATUS "Using Boost include dir at ${Boost_INCLUDE_DIR}")
+
+if(Boost_FOUND)
+  message(STATUS "Using Boost include dir at ${Boost_INCLUDE_DIR}")
+  message(STATUS "Using Boost lib version: ${Boost_LIB_VERSION}")
+  message(STATUS "Using Boost libs: ${Boost_LIBRARIES}")
+endif()
 
 ############
 # OpenSSL
 ############
 
-if (APPLE AND NOT IOS)
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fvisibility=default -std=c++14")
-  if (NOT OPENSSL_ROOT_DIR)
-      EXECUTE_PROCESS(COMMAND brew --prefix openssl
-        OUTPUT_VARIABLE OPENSSL_ROOT_DIR
-        OUTPUT_STRIP_TRAILING_WHITESPACE)
-    message(STATUS "Using OpenSSL found at ${OPENSSL_ROOT_DIR}")
+find_package(OpenSSL REQUIRED)
+
+if (WIN32)
+  list(APPEND OPENSSL_LIBRARIES ws2_32 crypt32 bcrypt)
+endif()
+
+if(OPENSSL_FOUND)
+  message(STATUS "Using OpenSSL include dir at ${OPENSSL_INCLUDE_DIR}")
+  message(STATUS "Using OpenSSL libs: ${OPENSSL_LIBRARIES}")
+endif()
+
+############
+# libsodium
+############
+
+find_library(SODIUM_LIBRARY sodium REQUIRED)
+if(SODIUM_LIBRARY)
+  message(STATUS "Using libsodium library at ${SODIUM_LIBRARY}")
+  find_path(SODIUM_INCLUDE_PATH sodium/crypto_verify_32.h)
+  if (SODIUM_INCLUDE_PATH)
+    message(STATUS "SODIUM_INCLUDE_PATH: ${SODIUM_INCLUDE_PATH}")
+  else()
+    message(FATAL_ERROR "Could not find required sodium/crypto_verify_32.h")
   endif()
 endif()
 
-find_package(OpenSSL REQUIRED)
-message(STATUS "Using OpenSSL include dir at ${OPENSSL_INCLUDE_DIR}")
+############
+# HIDAPI
+############
 
-if(STATIC AND NOT IOS)
-  if(UNIX)
-    set(OPENSSL_LIBRARIES "${OPENSSL_LIBRARIES};${CMAKE_DL_LIBS};${CMAKE_THREAD_LIBS_INIT}")
+option(USE_DEVICE_TREZOR "Trezor hardware wallet suport" ON)
```

Comment:
I get linker errors building with `USE_DEVICE_TREZOR=OFF` related to HIDAPI.

Can we remove this conditional and always find the package?

Reply:
@nsec1 any response?

Reply:
@woodser other than this comment, is this PR ready to merge?